src/main/java/org/opentripplanner/graph_builder/module/osm/WayProperties.java

```diff
@@ -33,14 +33,24 @@ public StreetTraversalPermission getPermission() {
   }
 
   public int hashCode() {
-    return bicycleSafetyFeatures.hashCode() + walkSafetyFeatures.hashCode() + permission.hashCode();
+    return (
+      (bicycleSafetyFeatures != null ? bicycleSafetyFeatures.hashCode() : 0) +
```

Comment:
In what conditions are these allowed to be null?

Reply:
The defaults are no longer set until they are needed because we need the speed for the way to set the defaults. I guess we could set the safety features with no values or some magic numbers as defaults before the actual defaults are set. This issue only happens in https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySet.java#L204-L208 as far as I know. I think we should guard against the NPEs anyway but should I also edit the log so it doesnt log the whole object?

Reply:
Use Objects.hashCode